wallet/db_sqlite3.c

```diff
@@ -24,8 +24,25 @@ static const char *db_sqlite3_fmt_error(struct db_stmt *stmt)
 
 static bool db_sqlite3_setup(struct db *db)
 {
+	char *filename;
 	sqlite3_stmt *stmt;
-	int err;
+	sqlite3 *sql;
+	int err, flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+
+	if (!strstarts(db->filename, "sqlite3://") || strlen(db->filename) < 10)
+		db_fatal("Could not parse the wallet DSN: %s", db->filename);
+
+	/* Strip the scheme from the dsn. */
+	filename = db->filename + strlen("sqlite3://");
```

Comment:
Wonder if we could use a `strafter` helper which did this dance for us?

Reply:
Sounds good, would you like 2 versions: one returning a pointer into the string and the other copying into a separate tallocated string?

I'll fix this up in my cleanups PR at the end of the week.